spec/index.html

```diff
@@ -684,12 +689,25 @@ <h2>Literals</h2>
         language tag MUST be well-formed according to
         <a data-cite="bcp47#section-2.2.9">section 2.2.9</a>
         of [[!BCP47]].</li>
+      <li>if and only if the <a>datatype IRI</a> is
+        <code>http://www.w3.org/1999/02/22-rdf-syntax-ns#dirLangString</code>,
+        a non-empty <a>language tag</a>
+        MUST be well-formed according to <a data-cite="bcp47#section-2.2.9">section 2.2.9</a>
+        of [[!BCP47]].
+        and <dfn>base direction</dfn> MUST be either `ltr` or `rtl`.</li>
     </ul>
 
     <p>A literal is a <dfn>language-tagged string</dfn> if the third element
-      is present. Lexical representations of language tags MAY be converted
-      to lower case. The value space of language tags is always in lower
-      case.</p>
+      is present and the fourth element is not present.
```

Comment:
I am a bit not at ease with this definition. Intuitively directional language-tagged strings are a special case of language-tagged string with an extra piece of information and this definition prevents to state that `rdf:dirLangString` is a sub-datatype of `rdf:langString`. It's why I would be personally more in favor of just adding an optional base direction to `rdf:langString`.

Reply:
We adding a new concept and it's about human language so it's not surprising there isn't a single, natural choice of design.

RDF itself doesn't have subclass/superclass. 

In XMLScheme there is a more complex system for derived datatypes - by extension and by restriction. `xsd:integer` is a restriction of `xsd:decimal`. A subdatatype for `rdf:dirLangString` is "by extension" adding a new facet which would be new to RDF core. Equally it could be a superdatatype with `rdf:langString` derived-by-restriction.

sub/super impacts the choice of `DATATYPE("abc"@en--rtl)` and `DATATYPE(?x) = rdf:langString`.

[Proposal for SPARQL functions](https://github.com/w3c/rdf-star-wg/wiki/Text-Direction-Proposal#functions)

The choices for SPARQL functions don't give a connection.
SKOS labels seem to be the most sensitive to the choice in RDF concepts and is also the sensitive to the SPARQL choices.

It seems to come down to a choice of no new datatype or a separate one.

The idea that there isn't a datatype/class for literals with base direction doesn't feel like RDF to me.
A toolkit can of course present the RDF in a convenient way but if the information isn't available (one datatype) it can't.

Reply:
Sorry, I missed your reply. Sorry for that.

> It seems to come down to a choice of no new datatype or a separate one.

Yes, I agree that having `rdf:dirLangString` is useful and we should have it.

> The choices for SPARQL functions don't give a connection.

I am not sure to fully understand what you are saying. Sorry for that. Isn't it already the case with XSD numeric datatypes? They extend one an other but `DATATYPE` only returns the most specific datatype (i.e. `DATATYPE("1"^^xsd:int) = xsd:int` and not `DATATYPE("1"^^xsd:int) = xsd:integer`)

> is "by extension" adding a new facet which would be new to RDF core.

Yes, that's true. However if we do not make `rdf:dirLangString` some kind of "subdatatype" of `rdf:langString` then it means for example that all vocabularies having `rdfs:range rdf:langString` would have to be updated to state that the range is the union of `rdf:langString` and `rdf:dirLangString`. This seems quite cumbersome to me.
I would tend to think that `"foo"^^en--ltr` entailing `"foo"^^en` is something we might want to have in some entailment regimes.
But it's probably a topic for  RDF-semantics or RDF-schema and not for RDF-concepts. So, I am convinced by the current PR. Thank you for your reply!